src/Microsoft.Fx.Portability/BreakingChangeParser.cs

```diff
@@ -245,6 +264,21 @@ private static void ParseNonStateChange(BreakingChange currentBreak, ParseState
                         currentBreak.SourceAnalyzerStatus = status;
                     }
                     break;
+                case ParseState.Categories:
+                    // Trim md list and code markers, as well as comment tags (in case the categories section is followed by a comment)
+                    var category = currentLine.Trim().TrimStart('*', '-', '!', '<', '>');
+                    if (string.IsNullOrWhiteSpace(category)) break;
+                    if (currentBreak.Categories == null)
+                    {
+                        currentBreak.Categories = new List<string>();
+                    }
+                    // If a list of allowed categories was provided, make sure that the category found is on the list
+                    if ( (!allowedCategories?.Contains(category, StringComparer.OrdinalIgnoreCase))?? false)
+                    {
+                        throw new InvalidOperationException($"Invalid category detected: {category}");
+                    }
+                    ((List<string>)currentBreak.Categories).Add(category);
```

Comment:
This explicit cast to `List<string>` isn't very safe since the interface is  `IEnumerable<T>`. We're assuming we know the underlying datatype. Can we change this to an `IList<T>`?

Can you fix `ApplicableApis` to `IList<T>` as well? [BreakingChangeParser.cs#L226](https://github.com/mjrousos/dotnet-apiport/blob/UpdateMDFiles/src/Microsoft.Fx.Portability/BreakingChangeParser.cs#L226)